### DISCUSSION TITLE
Updating `cacheControl` on existing images in storage

### DISCUSSION THREAD

**Discussion by wadclapp** [supabase/supabase#11746] [Questions]
I have probably like 1000 images in Storage.

How can I update the `cacheControl` property on them all (easily?)? They're currently all the default, which it seems is 60 minutes.

And if I upload from the Supabase dashboard, is there a way to change the default `cacheControl` value? I'd like it to be 1year (preferably depending on which bucket it's uploaded to).

The images shouldn't change, and from performance testing, I get the issue "Serve static assets with an efficient cache policy. A long cache lifetime can speed up repeat visits to your page." as the Cache TTL is 60 minutes on all my images.

Cheers

**Comment by wadclapp** [comment]
Found an issue related to this [here](https://github.com/supabase/storage-api/issues/270)

**Reply by liaujianjie** [reply]
Did you manage to find out?

**Reply by wadclapp** [reply]
Nope. Thanks for the answer. I plan on just waiting for the solution from Supabase, rather than writing some kind of script to re-upload them all, as I require the dashboard to upload the odd image still.

**Comment by liaujianjie** [comment]
Got this reply from the Supabase team:

> unfortunately not, you'll need to re-upload the asset at the moment, we have a burning task to implement an update method to allow updating the cache-control header via api

I guess not for now.

**Reply by kyoto21** [reply]
How about now? any news?

**Reply by ever-mira** [reply]
searching for the same.

**Reply by dendrofen** [reply]
available only with api for now...

**Reply by TejSwift** [reply]
Has this been updated now? Seems clumsy to not be able to change cacheControl after upload considering we get charged for exceeding limits.

**Reply by medi11** [reply]
same here, i already have +100k images and wanting to set the cache cache-control

**Reply by mcsrk** [reply]
any answer 👀

**Reply by aantti** [reply]
@itslenny
